scapy/layers/usb.py

```diff
@@ -284,3 +287,206 @@ def close(self):
             self.usbpcap_proc.kill()
 
     conf.USBsocket = USBpcapSocket
+
+
+def enumToScapyEnumDict(
+    en  # type: Enum
+):
+    return {el: el.name for el in en}
+
+
+def enumToScapyEnum(name, default, en, fmt):
+    return EnumField(name, default, enumToScapyEnumDict(en), fmt)
+
+
+class EndpointNumber(ByteField):
+    def m2i(self, pkt, x):
+        return cast(I, x)
+
+    @classmethod
+    def decode(
+        cls,
+        v  # type: int
+    ):
+        return bool(v >> 7), v & 0x7F
```

Comment:
This really isn't scapy-like. Have a look at [functions like `any2i`, `i2h`...](https://scapy.readthedocs.io/en/latest/build_dissect.html#manipulating-packets-manipulating-its-fields)